src2/tests.bash

```diff
@@ -32,49 +36,65 @@ echo "Setting server timeout to "$serverTime"sec"
 echo "Setting client timeout to "$clientTime"sec"
 echo "SERVER/CLIENT RUNNING ..."
 
-./Q1 -t $serverTime "$fifoname" > logs/q1.log 2> logs/q1.err &  # Un <-t nsecs> fifoname
+./Q2 -t $serverTime -l $nPlaces -n $nThreads "$fifoname" > logs/q2.log 2> logs/q2.err &  # Qn <-t nsecs> [-l nplaces] [-n nthreads] fifoname
 P1=$!
-./U1 -t $clientTime "$fifoname" > logs/u1.log 2> logs/u1.err &   # Qn <-t nsecs> [-l nplaces] [-n nthreads] fifoname
+./U2 -t $clientTime "$fifoname" > logs/u2.log 2> logs/u2.err &   # Un <-t nsecs> fifoname
 P2=$!
 wait $P1 $P2
 echo "END OF SERVER/CLIENT"
 
-cd logs || exit
+cd logs
+nENTER=`grep ENTER q2.log | wc -l`
+nIWANT=`grep IWANT u2.log | wc -l`
+nFAILD=`grep FAILD u2.log | wc -l`
+nGAVUP=`grep GAVUP q2.log | wc -l`
+n2LATE=`grep 2LATE q2.log | wc -l`
+nCLOSD=`grep CLOSD u2.log | wc -l`
+nIAMIN=`grep IAMIN u2.log | wc -l`
+nTIMUP=`grep TIMUP q2.log | wc -l`
+nRECVD=`grep RECVD q2.log | wc -l`
 
-nREQST=`grep IWANT u1.log | wc -l`
-nFAILD=`grep FAILD u1.log | wc -l`
-nGAVUP=`grep GAVUP q1.log | wc -l`
+echo "--SERVER--"
+echo "RECVD: $nRECVD"
+echo "ENTER: $nENTER"
+echo "GAVUP: $nGAVUP"
+echo "TIMUP: $nTIMUP"
+echo "2LATE: $n2LATE"
 
-n2LATE=`grep 2LATE q1.log | wc -l`
-nCLOSD=`grep CLOSD u1.log | wc -l`
-
-nIAMIN=`grep IAMIN u1.log | wc -l`
-nENTER=`grep ENTER q1.log | wc -l`
-
-echo "Requests sent: $nREQST"
-echo "Failed requests: $nFAILD"
-echo "Gave up requests: $nGAVUP"
+echo "--CLIENT--"
+echo "IWANT: $nIWANT"
+echo "IAMIN: $nIAMIN"
+echo "CLOSD: $nCLOSD"
+echo "FAILD: $nFAILD"
 
 valid1=0;
 valid2=0;
+valid3=0;
 
-if  [ $n2LATE -eq $nCLOSD ] ; then
-  echo -e "${GREEN}[PASSED] ${NC}2LATE - too late requests: $n2LATE"
+if  [ $nENTER -eq $nIAMIN ] ; then
+  echo -e "${GREEN}[PASSED] ${NC}ENTER - accepted requests: $nENTER"
   valid1=1;
 else
-  echo -e "${RED}[FAILED] ${NC}2LATE"
+  echo -e "${RED}[FAILED] ${NC}ENTER"
 fi
 
-if  [ $nIAMIN -eq $nENTER ] ; then
-  echo -e "${GREEN}[PASSED] ${NC}ENTER - accepted requests: $nENTER"
+if  [ $nENTER -eq $nTIMUP ] ; then
+  echo -e "${GREEN}[PASSED] ${NC}TIMEUP - number of timeup's: $nTIMUP"
   valid2=1;
 else
-  echo -e "${RED}[FAILED] ${NC}ENTER"
+  echo -e "${RED}[FAILED] ${NC}TIMEUP"
+fi
+
+if  [ $nFAILD -eq "$(echo "$nIWANT - ($nENTER + $n2LATE)" | bc)" ] ; then
```

Comment:
Explica-me este teste sff. O numero de FAILD tem de ser igual à diferença entre o nº de IWANT's e a soma de ENTER's com 2LATE's o.O
E o que é o pipe para bc?

Reply:
O bc serve para fazer contas na command line
Quando o cliente faz um request IWANT há 3 opções, ou dá ENTER ou dá 2LATE ou dá FAILD, logo se não for ENTER OU 2LATE vai ser FAILD

Reply:
Estava a escrever o relatório e estive a pensar, então concordas com isto:

---
3. O número de **FAILD** tem de ser igual à diferença entre o número de **IWANT** e a soma do número de **ENTER** com **2LATE** - A thread cliente pode tomar 2 caminhos, que dependem se ela recebe resposta a partir do fifo privado ou não. De todos os pedidos **IWANT**, alguns resultarão em **IAMIN** e outros em **FAILD**. o número de **FAILD** será então o número de pedidos que não receberam a resposta **ENTER** nem **2LATE**.

---